instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -83,23 +85,52 @@
   }
 
   /**
-   * Returns a new client {@link Instrumenter} which will create client spans and inject context
-   * into requests.
+   * Sets the {@link StartTimeExtractor} and the {@link EndTimeExtractor} to extract the timestamp
+   * marking the start and end of processing.
+   */
+  public InstrumenterBuilder<REQUEST, RESPONSE> setTimeExtractors(
+      StartTimeExtractor<REQUEST> startTimeExtractor, EndTimeExtractor<RESPONSE> endTimeExtractor) {
```

Comment:
These two should probably be always set together - start & end timestamps need to be comparable, so it does not make any sense to set just one and compare with the SDK timestamp.